Firmware/FFBoard/Src/SerialFFB.cpp

```diff
@@ -74,7 +74,7 @@ void SerialFFB::set_gain(uint8_t gain){
  * Returns the index where the effect was created or -1 if it can not be created
  */
 int32_t SerialFFB::newEffect(uint8_t effectType){
-	uint32_t idx = this->effects_calc->find_free_effect(effectType);
+	int32_t idx = this->effects_calc->find_free_effect(effectType);
```

Comment:
@Ultrawipf Here is another similar question that may be more important.

Reply:
As the function can return -1 on error thats also required to be signed. It never happens because enough effects can be created for any game i have ever seen but in theory it would be an issue if you create too many with the serial ffb mode.